### ISSUE TITLE
spawn error with new client release

### ISSUE THREAD

**Issue by aprabhak2** [genome/civic-client#139]
@jmcmichael: I just tried a fresh start with your latest release, and running into issues with npm install and finally with gulp serve. Could you kindly take a look at this and let me know what i am missing? 
The node version I use is v0.12.2 and npm is 2.10.1. 

![image](https://cloud.githubusercontent.com/assets/10712456/8260641/d9f6a3e2-1679-11e5-83e5-195a441be153.png)

**Comment by jmcmichael** [comment]
This looks like it's caused by a recently-fixed bug in the optipng package (imagemin/optipng-bin@f14d05a). I just tried an install in a fresh checkout of the client, and ran the gulp serve task, so I think that this update has been applied to the npm module. If you try the following on your current install, it may fix the issue, and also re-installing from a fresh checkout should work too:

```bash
npm cache clean gulp-imagemin
rm -rf node_modules/gulp-imagemin
npm install gulp-imagemin
```

**Comment by aprabhak2** [comment]
@jmcmichael, I tried your suggestions above on the existing directory by cleaning cache, removing node modules. Same error. Also tried removing npm-global(which we use as prefix in .npmrc) and did a fresh checkout of the client and tried again. The errors are still persisting. Attached all the error snapshots we see when running npm install. While npm install doesn't explicitly complain, gulp serve doesn't start due to these errors. 

Any suggestions? 
 
![gifsicle_error](https://cloud.githubusercontent.com/assets/10712456/8310385/64441810-1984-11e5-8223-a524c2a082d6.JPG)
![jpegtran_error](https://cloud.githubusercontent.com/assets/10712456/8310384/64411192-1984-11e5-83c0-4b9da883a753.JPG)
![optipng_error](https://cloud.githubusercontent.com/assets/10712456/8310383/6440e924-1984-11e5-8bfa-1b9b681f0a05.JPG)
![pngquant_error](https://cloud.githubusercontent.com/assets/10712456/8310386/6444f104-1984-11e5-8a97-5da91b05db32.JPG)

**Comment by bwalsh** [comment]
As an _experiment_, I created the following docker file to install the civic-client on centos7
For some reason `npm install` would hang under centos6.6 (at least on macos boot2docker)
Hope this helps,
-b

```
FROM centos:latest

# install node
RUN yum -y install git  libpng-devel  wget curl tar bzip2

# install pre-built latest node
RUN wget https://nodejs.org/dist/v0.12.5/node-v0.12.5-linux-x64.tar.gz
RUN tar --strip-components 1 -xzvf node-v* -C /usr/local

# get the app
RUN git clone https://github.com/genome/civic-client.git
WORKDIR civic-client

# since by default, there is only one user 'root' in a container
# we need to allow root to do the bower install 
RUN echo '{ "allow_root": true }' > /root/.bowerrc

# install bower and gulp globally
RUN  npm install --allow-root  -g bower gulp   

# install the remainder of the modules in package.json
RUN  npm install --allow-root  

# install bower dependencies
RUN bower install  --allow-root    

# allow access to port 3001
EXPOSE  3001

# run our application
CMD ["gulp", "serve"]
```

**Comment by jmcmichael** [comment]
@aprabhak2, it looks like all of those error message have to do with fetching binaries from github to power various features of the imagemin component. I was able to successfully retrieve one of the binaries (https://raw.github.com/imagemin/pngquant-bin/v3.0.0/vendor/linux/x64/pngquant) from my location, so I believe that the URLs npm is using during the gulp-imagemin installation are correct.

Are you behind a firewall that might prohibit certain types of binaries from being fetched?

**Comment by jmcmichael** [comment]
Also, I'm running node v0.12.0. You may want to try down-grading to v0.12.0 to see if you problems persist. The [Node Version Manager](https://github.com/creationix/nvm) can be very helpful in these kinds of situations.

**Comment by aprabhak2** [comment]
@jmcmichael 
Sorry for not responding sooner, this week's been busy. 

I did a cache clean and started with a new clone and while the above npm install errors still persists(not sure what more settings we need to enable behind the firewall), this time around the gulp serve did not throw any error. 

Also, in another install (since the request to the raw.github.com failed, and it is unable to find packages in sourceforge), I was able able to manually compile these packages and link the binary to the vendor folder. It is not ideal, but it works. 

As far as reasons why it all of a sudden works, these may have played a role?:
Looks like imagemin’s latest commit was 4 days ago - https://github.com/imagemin/imagemin
Gulp-imagemin’s latest commit was 5 days ago - https://github.com/sindresorhus/gulp-imagemin
Optipng-bin’s latest commit was 5 days ago - https://github.com/imagemin/optipng-bin

Maybe they found some bugs in the last few days and fixed them?
